flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateRequestClassifier.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.forst;
+
+import org.apache.flink.runtime.asyncprocessing.StateRequest;
+import org.apache.flink.runtime.asyncprocessing.StateRequestContainer;
+import org.apache.flink.runtime.asyncprocessing.StateRequestType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The ForSt {@link StateRequestContainer} which can classify the state requests by ForStDB
+ * requestType (Get、Put or Iterator).
+ */
+public class ForStStateRequestClassifier implements StateRequestContainer {
+
+    private final List<ForStDBGetRequest<?, ?>> dbGetRequests;
+
+    private final List<ForStDBPutRequest<?, ?>> dbPutRequests;
+
+    public ForStStateRequestClassifier() {
+        this.dbGetRequests = new ArrayList<>();
+        this.dbPutRequests = new ArrayList<>();
+    }
+
+    @Override
+    public void offer(StateRequest<?, ?, ?> stateRequest) {
+        if (stateRequest.getState() == null) {
+            // The state could be null only for {@link StateRequestType#SYNC_POINT}. The
+            // ForStateBackend does nothing for {@link StateRequestType#SYNC_POINT}.
+            return;
+        }
+        convertStateRequestsToForStDBRequests(stateRequest);
```

Comment:
I'd suggest a lazy-initialization of classifying these requests into put or get. For example, when calling `pollDbGetRequests` or `pollDbPutRequests`? We'd better unburden the task thread from these computation.

Reply:
In my opinion, classifying the state requests at the time of `StateRequestContainer#offer` can reduce the total overhead of putting/iterating state request,  although it will increase the burden of the task thread. 

I think no matter which way is chosen, the impact on the system is very very small and insignificant. Maybe we can still use the current way, and we can judge whether we need to change it according to the benchmark result in the future？

WDYT?

Reply:
Agreed. Considering the overhead is small, we could optimize this later if this is a big one.